src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs

```diff
@@ -292,7 +292,12 @@ private static ImmutableArray<ReferenceDirective> GetReferenceDirectives(Compila
                     continue;
                 }
 
-                result |= QuickAttributeHelpers.GetQuickAttributes(directive.Name.GetUnqualifiedName().Identifier.ValueText, inAttribute: false);
+                if (directive.Name is not NameSyntax name)
+                {
+                    continue;
```

Comment:
I didn't follow this. Why is it okay to bail out here?

Reply:
So QuickAttributeChecker is a helper type that allows the compiler to trivially bail out whne looking for attributes *by name*.  Effectively, it keeps track of hte identifiers used in attributes (including seeing through using aliases).

However, because this is *only* for attributes, the only sorts of aliases that are relevant are thsoe that point at real names.  

e.g. if i have:

```
using X = System.Runtime.CompilerServices.TypeForwardedToAttribute;
```

then that is relevant later on if i see `[X]`.  However, if i have `using X = (int, int);` that's never interesting because it would always be illegal for `[X]` as attributes can only referenced named items.

Put another way, we're building a map of `X -> TypeForwardedToAttribute` so that hte compiler can quickly *syntactically* determine "Hey!  I have 'X', is it worth binding if i'm looking for TypeForwardedTo?".   Since hte compiler is only doing this lookup for named entities that appear in attribute, using-aliases themselves are only interesting if they alias something that could be in an attribute (named a named type).